interactors/drop_block_interactor.py

```diff
@@ -0,0 +1,41 @@
+import os
+
+from entities.gameboard_entity import Gameboard
+from entities.block_entity import Block
+
+class DropBlockInteractor():
+    def __init__(self, gameboard: Gameboard):
+        self.gameboard = gameboard
+
+    def drop_block(self, block: Block):
+
+        # Check if any move is possible
+        can_move = True
+        while can_move:
+
+            lowest_active_blocks = []
+            for j in range(self.gameboard.game_matrix_width):
+                for i in reversed(range(self.gameboard.game_matrix_height)):
+                    if self.gameboard.game_matrix[i][j][2] == 'a':
+                        lowest_active_blocks.append((i, j))
+                        break  # Break the inner loop when you find an active block 
+            can_move = all(i + 1 < self.gameboard.game_matrix_height and self.gameboard.game_matrix[i + 1][j][0] == 0
```

Comment:
but add some empty line here :D (to separate for from next instruction)

Reply:
done